tools/cli/NearObjectCli.cxx

```diff
@@ -231,6 +231,42 @@ AddEnumOption(CLI::App* app, std::optional<EnumType>& assignTo, bool isMandatory
     }
 }
 
+/**
+ * @brief Displays and returns application configuration parameters.
+ *
+ * @param applicationConfigurationParameterData The possible application configuration parameters that may be set.
+ * @return std::vector<uwb::protocol::fira::UwbApplicationConfigurationParameter>.
+ */
+std::vector<uwb::protocol::fira::UwbApplicationConfigurationParameter>
+ProcessApplicationConfigurationParameters(UwbApplicationConfigurationParameterData applicationConfigurationParameterData)
+{
+    std::vector<uwb::protocol::fira::UwbApplicationConfigurationParameter> applicationConfigurationParameters;
+
+    std::cout << "Selected parameters:" << std::endl;
+    for (const auto& [applicationConfigurationParameter, applicationConfigurationParameterValue] : applicationConfigurationParameterData.GetValueMap()) {
+        std::visit([&applicationConfigurationParameter, &applicationConfigurationParameters](auto&& arg)
+        {
+            applicationConfigurationParameters.push_back({ applicationConfigurationParameter, arg });
+
+            using ParameterValueT = std::decay_t<decltype(arg)>;
+            if constexpr (std::is_enum_v<ParameterValueT>) {
+                std::cout << magic_enum::enum_name(applicationConfigurationParameter) << "::" << magic_enum::enum_name(arg) << std::endl;
+            } else if constexpr (std::is_same_v<ParameterValueT, uint8_t>) {
+                std::cout << magic_enum::enum_name(applicationConfigurationParameter) << "::" << static_cast<int>(arg) << std::endl;
```

Comment:
Even though it's not an issue in this instance, we should use unary integer promotion here instead as good practice:

```suggestion
                std::cout << magic_enum::enum_name(applicationConfigurationParameter) << "::" << +arg << std::endl;
```

The presence of the cast might create a tendency to do that elsewhere in cases where a narrowing conversion would occur, and truncate data.